src/matlab2tikz.m

```diff
@@ -1519,33 +1539,29 @@ case guitypes()
     % a reasonable default.
     matlabDefaultNumMinorTicks = 3;
     if ~isempty(ticks)
-        options = opts_add(options, ...
-            [axis,'tick'], sprintf('{%s}', ticks));
+        options = opts_add(options, [axis,'tick'], sprintf('{%s}', ticks));
```

Comment:
So i would aggree to change it as you have and clean up the rest when someone has some spare time.